### PR TITLE
Origin/dev

### PR DESCRIPTION
**Pull request by Machtay**
I fixed the issue with the loop that was stalling the run. I did a test run last night and just reviewed  the results and everything looks good. You can see the test run results here if you want to check it out: 

`/fs/ess/PAS1960/BiconeEvolutionOSC/BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Run_Outputs/2022_12_29_Test`

Note that the fitness scores are nonsense because so few neutrinos were thrown. But, the point of the test was only to resolve the issue where gain files read into AraSim weren't being created. You can view those files (they are .txt files) here: 

`/fs/ess/PAS1960/BiconeEvolutionOSC/BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Run_Outputs/2022_12_29_Test/2022_12_29_Test.xf`

Let me know if you notice anything wrong or need a quick fix before I make this merge. Once it's made, I'll begin a new run.

### CODE REVIEW

**Review by JulieRolla — APPROVED** [review]

File: BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoARA.py [hunk 1]
```diff
@@ -59,11 +59,11 @@ def readFile(indiv, freqNum):
     n = 37
     m = 73
     #uanName = uanLoc+str(indiv)+"_"+str(freqNum)+".uan"
-    uanName = g.WorkingDir + "/" + "Run_Outputs" + "/" +g.RunName+ "/" + str(g.gen) + "_" + str(indiv)+"_"+str(freqNum)+".uan" #changed from g.indiv + 1 to just indiv 
+    uanName = g.WorkingDir + "/" + "Run_Outputs" + "/" +g.RunName+ "/uan_files/" + str(g.gen) + "_" + str(indiv)+"_"+str(freqNum)+".uan" #changed from g.indiv + 1 to just indiv
```
Comment (ezio-melotti) [inline BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoARA.py:62]:
```suggestion
    uanName = f'{g.WorkingDir}/Run_Outputs/{g.RunName}/uan_files/{g.gen}_{indiv}_{freqNum}.uan'  #changed from g.indiv + 1 to just indiv 
```

File: BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoARA2.py [hunk 2]
```diff
@@ -59,11 +59,12 @@ def readFile(indiv, freqNum):
     n = 37
     m = 73
     #uanName = uanLoc+str(indiv)+"_"+str(freqNum)+".uan"
-    uanName = g.WorkingDir + "/" + "Run_Outputs" + "/" +g.RunName+ "/" + str(g.gen) + "_" + str(indiv)+"_"+str(freqNum)+".uan" #changed from g.indiv + 1 to just indiv 
+    uanName = g.WorkingDir + "/" + "Run_Outputs" + "/" +g.RunName+ "/uan_files/" + str(g.gen) + "_uan_files/" + str(g.gen) + "_" + str(indiv)+"_"+str(freqNum)+".uan" #changed from g.indiv + 1 to just indiv
```
Comment (ezio-melotti) [inline BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoARA2.py:62]:
```suggestion
    uanName = f'{g.WorkingDir}/Run_Outputs/{g.RunName}/uan_files/{g.gen}_uan_files/{g.gen}_{indiv}_{freqNum}.uan'  #changed from g.indiv + 1 to just indiv 
```

**Comment by ezio-melotti** [comment]
An even better solution for handling the path would be to use `pathlib.Path` instead of `str` in
https://github.com/osu-particle-astrophysics/GENETIS-ARA/blob/d63c6958cd500c43cd72031de47100abc35b4a28/BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoARA.py#L94

and then do:
```py
uanName = g.WorkingDir / 'Run_Outputs' / g.RunName / 'uan_files' / f'{g.gen}_{indiv}_{freqNum}.uan'
```

Also why there are two copies of the file (`XFintoARA.py` and `XFintoARA2.py`)?

**Comment by Machtay** [comment]
> An even better solution for handling the path would be to use `pathlib.Path` instead of `str` in
> 
> https://github.com/osu-particle-astrophysics/GENETIS-ARA/blob/d63c6958cd500c43cd72031de47100abc35b4a28/BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoARA.py#L94
> 
> and then do:
> 
> ```python
> uanName = g.WorkingDir / 'Run_Outputs' / g.RunName / 'uan_files' / f'{g.gen}_{indiv}_{freqNum}.uan'
> ```
> 

Thanks for pointing this out! I'll keep things the way they are for now but make this a part of a larger commit after the upcoming run to clean up all of our python scripts that handle paths the same way. I think it will clean up those scripts a lot!

> Also why there are two copies of the file (`XFintoARA.py` and `XFintoARA2.py`)?

Yeah I noticed this too but didn't take too close of a look at the differences since it seems like it isn't being used. It looks like we may have been trying to make a script for using more frequencies than we typically use (since it has `numFreq=68` instead of `numFreq=60`). If this is something we want to do, we can make that a variable in the loop and read it into the script as an argument (and we might need to modify the frequency list too then). After this run I'll remove that script and update XFintoARA.py to use a variable `numFreq` if we think that's useful.

**Pull request merged**